src/duckling/migration/map-migration.ts

```diff
@@ -3,7 +3,7 @@ import {MapVersion, compareVersions, versionCompareFunction, VersionCompatibilit
  * Represents a single migration that needs to be run when updating to the specified version.
  */
 export interface MapMigration {
-    type : "code" | "editor-version"; // | "editor-built-in",
+    type : "code" | "editor-version" | "existing-code", // "editor-version" is really "engine-version"
```

Comment:
The only difference between the types is how the editor identifies the migration. Instead of having a type, which confuses everybody (my bad), we should have different fields for the different identifier types. Example: "name" for "editor-version" and "existing-code". The "code" migration type should have a "path" identifier.